### PR TITLE
[WIP] Objectifying $kga

### PR DESCRIPTION
**Pull request by kevinpapst**
Changes proposed in this pull request:
- Large refactoring of the $kga to an ArrayObject, allowing backward compatibility with the array usage
- The new object structure allows for OO programming
- Moved Authenticator to Registry (fixes kimai/roadmap#3)
- Rewritten Debug extension to not-preload data, but use JS always / allow for easier debugging the kga object / removed reload button / removed unused images
- added Kimai_Translation_Service and Kimai_Translation_Data, preventing multiple unnecessary data loads
- rewritten config loading, utilizing the new Config ArrayObjects (preventing a lot of duplicated database requests)
- removed vars.php - logic went into new Config objects
- worked on some translations
- code formattings on the fly
- fixed editLimit check in editing and creating expenses

Reason for this pull request:
- Future improvements of Kimai and allowing OO programming

This PR is to be discussed and needs some tweaks, I added it for discussion/getting feedback at this point

### CODE REVIEW

File: libraries/Kimai/Auth/Abstract.php [hunk 1]
```diff
@@ -58,9 +58,9 @@ public function __construct($database = null, $kga = null)
     }
 
     /**
-     * @param array $kga
+     * @param $kga
```
Comment (simonschaufi) [inline libraries/Kimai/Auth/Abstract.php]: specify type

File: libraries/Kimai/Database/Mysql.php [hunk 2]
```diff
@@ -2807,34 +2778,34 @@ public function checkUserInternal($kimai_user)
...
     /**
-     * write global configuration into $this->kga including defaults for user settings.
+     * Returns all configuration variables
      *
-     * @return array $this->kga
-     * @author th
+     * @return array with the options from the configuration table
      */
-    public function get_global_config()
+    public function configuration_get_data()
```
Comment (simonschaufi) [inline libraries/Kimai/Database/Mysql.php]: rename to getConfigurationData

File: libraries/Kimai/Database/Mysql.php [hunk 3]
```diff
@@ -2807,34 +2778,34 @@ public function checkUserInternal($kimai_user)
...
-        $this->kga['conf']['hideClearedEntries'] = 0;
+        return $config_data;
+    }
+
+    /**
+     * Return all available status entries.
+     *
+     * @return array
+     */
+    public function status_get_all()
```
Comment (simonschaufi) [inline libraries/Kimai/Database/Mysql.php]: getAllStatus

**Comment by simonschaufi** [comment]
rebase your branch on top of upstream/develop branch to make it mergable again

**Comment by kevinpapst** [comment]
Yes! But let me finish my work first ;-) I'll let you know when I am "done" with all the refactoring.

**Comment by simonschaufi** [comment]
why?

**Comment by kevinpapst** [comment]
Because I cannot rebase properly a previously pushed branch.
Also there were multiple merge conflicts, so I deleted it.
Will push it again, once its truely ready for feedback and merging.
Due to the amount of changes we should not merge between my final PR and the merge.

**Comment by simonschaufi** [comment]
rebasing is always possible, i just rebased a branch from last year. of course it takes some time resolving the merge conflicts but it works ;) the worst commit was the move of all files one folder up (removing the core folder)

**Comment by kevinpapst** [comment]
then phpstorm is a bitch, it won't let me push a rebased branch, but wants that I merge again.
So I aborted the rebase and deleted the branch ;-)

**Comment by simonschaufi** [comment]
if you just do a rebase, you have force push because you are rewriting git history. I always do that force push on the command line.

**Pull request closed without merging**